nengo/neurons.py

```diff
@@ -34,19 +34,37 @@ def _argreprs(self):
     def current(self, x, gain, bias):
         """Compute current injected in each neuron given input, gain and bias.
 
+        Note that ``x`` is assumed to be already projected onto the encoders
+        associated with the neurons and normalized to radius 1, so the maximum
+        expected current for a neuron occurs when input for that neuron is 1.
+
         Parameters
         ----------
-        x : (n_neurons,) array_like
-            Vector-space input.
+        x : (n_samples,) or (n_samples, n_neurons) array_like
+            Scalar inputs for which to calculate current.
         gain : (n_neurons,) array_like
             Gains associated with each neuron.
         bias : (n_neurons,) array_like
             Bias current associated with each neuron.
+
+        Returns
+        -------
+        current : (n_samples, n_neurons)
+            Current to be injected in each neuron.
+            If ``n_samples`` is 1, the extra dimension will be removed.
         """
         x = np.array(x, dtype=float, copy=False, ndmin=1)
         gain = np.array(gain, dtype=float, copy=False, ndmin=1)
         bias = np.array(bias, dtype=float, copy=False, ndmin=1)
-        return gain * x + bias
+
+        if x.ndim == 1:
+            x = x[:, np.newaxis]
+        elif x.ndim > 1 and x.shape[1] != gain.shape[0]:
+            raise ValidationError("Expected shape (%d, %d); got (%d, %d)."
+                                  % ((x.shape[0], gain.shape[0]) + x.shape),
```

Comment:
This string formatting will break if `x.ndim > 2`.

Reply:
Do you think we should disallow that case? Previously it would "just work" according to numpy's usual broadcasting, but I can see an argument for limiting things to the typical use-cases.

Reply:
Actually it would only work if we required `x.shape[-1]` to be equal to `gain.shape[0]`. Even though I really prefer more general code, it's probably better to disallow it. That way, we can keep the docstring as-is and have it properly documented.

Reply:
This is done in option 3.